### ISSUE TITLE
Direct reply improvements

### ISSUE THREAD

**Issue by Maxr1998** [wasdennnoch/AndroidN-ify#756]
Direct reply still has some room to improve, this will serve as my to-do list:
- ~~Animate in the input layout with `createCircularReveal()`~~
- ~~Tint send button accordingly (disable and grey out when no input)~~
- ~~Tint text cursor accordingly~~
- ~~Scroll to the notification before opening keyboard~~
- ~~Extract RemoteInput objects from a WearableExtender~~
- ~~Also extract RemoteInput from CarExtender~~
- ~~Fix HeadsUp not shown long enough to reply~~
- Auto remove notification after timeout

**Comment by wasdennnoch** [comment]
I can only assign collaborators apparently.

**Comment by Maxr1998** [comment]
Ok, thanks!

**Comment by Maxr1998** [comment]
List updated according to #839.

**Comment by wasdennnoch** [comment]
Found a little problem, the bottom stack margin cuts the RemoteInput when the keyboard opens and there are enough/large notifications: https://www.youtube.com/watch?v=cmisvrZVu8w

**Comment by Maxr1998** [comment]
Experienced the same, hopefully the still-to-be-implemented code to autoscroll and lock to position fixes the issue.

**Comment by wasdennnoch** [comment]
Crash when swiping away heads-up with RemoteInput active:
```

09-20 21:57:54.445 6212-6212/com.android.systemui E/AndroidRuntime: FATAL EXCEPTION: main
                                                                    Process: com.android.systemui, PID: 6212
                                                                    Theme: themes:{default=overlay:com.resurrectionremix.pitchblack.solar, fontPkg:com.resurrectionremix.pitchblack.solar, com.android.systemui=overlay:com.resurrectionremix.pitchblack.solar, com.android.systemui.navbar=overlay:com.resurrectionremix.pitchblack.solar}
                                                                    java.lang.NullPointerException: Attempt to read from field 'com.android.systemui.statusbar.ExpandableNotificationRow com.android.systemui.statusbar.NotificationData$Entry.row' on a null object reference
                                                                        at com.android.systemui.statusbar.stack.AmbientState.getTopHeadsUpEntry(AmbientState.java:185)
                                                                        at com.android.systemui.statusbar.stack.AmbientState.getTopHeadsUpPushIn(AmbientState.java:162)
                                                                        at com.android.systemui.statusbar.stack.AmbientState.getInnerHeight(AmbientState.java:158)
                                                                        at com.android.systemui.statusbar.stack.StackScrollAlgorithm.findNumberOfItemsInTopStackAndUpdateState(StackScrollAlgorithm.java:744)
                                                                        at com.android.systemui.statusbar.stack.StackScrollAlgorithm.getStackScrollState(StackScrollAlgorithm.java:172)
                                                                        at com.android.systemui.statusbar.stack.NotificationStackScrollLayout.updateChildren(NotificationStackScrollLayout.java:443)
                                                                        at com.android.systemui.statusbar.stack.NotificationStackScrollLayout.-wrap1(NotificationStackScrollLayout.java)
                                                                        at com.android.systemui.statusbar.stack.NotificationStackScrollLayout$1.onPreDraw(NotificationStackScrollLayout.java:220)
                                                                        at android.view.ViewTreeObserver.dispatchOnPreDraw(ViewTreeObserver.java:944)
                                                                        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2063)
                                                                        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1115)
                                                                        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6023)
                                                                        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:858)
                                                                        at android.view.Choreographer.doCallbacks(Choreographer.java:670)
                                                                        at android.view.Choreographer.doFrame(Choreographer.java:606)
                                                                        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:844)
                                                                        at android.os.Handler.handleCallback(Handler.java:739)
                                                                        at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                        at android.os.Looper.loop(Looper.java:148)
                                                                        at android.app.ActivityThread.main(ActivityThread.java:5471)
                                                                        at java.lang.reflect.Method.invoke(Native Method)
                                                                        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                                        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
                                                                        at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:132)
```

**Comment by Maxr1998** [comment]
Thanks, I'm going to look at it.

**Comment by Maxr1998** [comment]
Fixed in 139e1740a21f12acff6493acf9fa0fcd0765170a.

**Comment by wasdennnoch** [comment]
I guess this can be closed.
